app.go

```diff
@@ -193,9 +198,55 @@ func New(opts ...Option) *App {
 		app.provide(p)
 	}
 	app.provide(func() Lifecycle { return app.lifecycle })
+
+	if app.err != nil {
+		app.logger.Printf("Error after options were applied: %v", app.err)
+		return app
+	}
+
+	if err := app.executeInvokes(); err != nil {
+		app.err = err
+	}
+
 	return app
 }
 
+// Err returns an error that may have been encountered during the
+// graph resolution.
+//
+// This includes things like incomplete graphs, circular dependencies,
+// missing dependencies, invalid constructors, etc.
+func (app *App) Err() error {
+	return app.err
+}
+
+// Execute invokes in order supplied to New.
+//
+// It might be worthwhile to consider adding context.Context to this function
+// so we can handle the infinite-invokes.
+//
+// Returns the first error encountered. Maybe shoul multiErr them together.
+func (app *App) executeInvokes() error {
+	var err error
+
+	for _, fn := range app.invokes {
+		fname := fxreflect.FuncName(fn)
+		app.logger.Printf("INVOKE\t\t%s", fname)
+
+		if _, ok := fn.(Option); ok {
+			err = fmt.Errorf("fx.Option should be passed to fx.New directly, not to fx.Invoke: fx.Invoke received %v", fn)
+		} else {
+			err = app.container.Invoke(fn)
+		}
+
+		if err != nil {
+			app.logger.Printf("Error during %q invoke: %v", fname, err)
```

Comment:
This is bugged, you need a break here. Otherwise the next invoke will just nuke the err.